cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -237,27 +242,26 @@ func (k *proxy) ListResources(labels map[string]string, namespaces ...string) ([
 		return nil, errors.Wrap(err, "failed to list api resources")
 	}
 
-	// If labels indicates that resources of a specific provider should be listed, exclude CRDs of other providers.
+	// Exclude from discovery the objects from the cert-manager/provider's CRDs.
```

Comment:
To confirm my understanding.

We're calling this func in two cases:
* certManagerClient.EnsureLatestVersion
    * in this case we're not listing any CRDs
* providerComponents.Delete
     * in this case we're not listing any provider CRDs, i.e. we still list Certificate and CertificateRequest (which is what we want)

If I'm correct, maybe we should adjust the godocs of the func slightly, as they could also be interpreted differently (i.e. that the func is only not returning resources of the CRDs of the *current* component)

Reply:
~ right
In both cases we are not listing resources for any CRDs installed by clusterctl (including cert-manager CRDs and providers CRDs). 

In other words, we are only concerned about provider components (deployments, RBAC rules, CRD definitions) but not in the resource created on top of that (Clusters, Machines, AWSCluster etc) because the latter are not relevant for upgrade/delete.